### DISCUSSION TITLE
missing_context_reference warning - what does this mean?

### DISCUSSION THREAD

**Discussion by katecrombie** [NASA-PDS/validate#505] [Q&A] [answered]
I have a large bundle that I validated with the latest version of validate and got the following summary:

 0 error(s)
  306271 warning(s)

  Product Validation Summary:
    305971     product(s) passed
    0          product(s) failed
    0          product(s) skipped

  Referential Integrity Check Summary:
    305971     check(s) passed
    0          check(s) failed
    0          check(s) skipped

  Message Types:
    305951       warning.integrity.missing_context_reference
    320          warning.integrity.member_not_found

We understand and have fixed the  warning.integrity.member_not_found.  What we don't know how to fix are the  warning.integrity.missing_context_reference products.  The full message is as follows:

WARNING  [warning.integrity.missing_context_reference]   The context reference 'urn:nasa:pds:sci_calibrated:data:xxx_a' could not be found in this bundle but it was defined in urn:nasa:pds:sci_calibrated:browse:xxx_a::2.1.

If I manually look  for the data collection product, I can find it. It has a lid that matches the lid that supposedly could not be found. I don't understand what the warning message is trying to tell me.

Help?

**Comment by jordanpadams** [comment]
@katecrombie what version of validate are you running? I believe we reverted that change before we tagged a stable release but may have missed something.

**Comment by katecrombie** [comment]
We ran using version 2.1.0

PDS Validate Tool Report

Configuration:
   Version                       2.1.0
   Date                          2022-05-26T13:49:27Z

**Comment by jordanpadams** [comment, marked as answer]
@katecrombie can you try the latest release? https://github.com/NASA-PDS/validate/releases/tag/v2.2.0

**Comment by katecrombie** [comment]
@jordanpadams we'll give it try.  It will be a while before we get results to share.

**Comment by katecrombie** [comment]
@jordanpadams validate version 2.2.0 "fixed" the issue.  We now validate with no errors (after a fix) and no warnings.

**Comment by jordanpadams** [comment]
@katecrombie excellent! for some background, that WARNING you were seeing was some new functionality we attempted to implement to ensure that parent bundle/collections contained the context references (targets, instruments, etc.) of their child products (#69) . However, the implementation was (1) riddled with bugs and (2) the error message was obviously not intuitive enough. We may implement this again in the future to enhance search, but it has not bubbled up just yet.
